apps/meteor/client/providers/ImageGalleryProvider.tsx

```diff
@@ -23,8 +23,10 @@ const ImageGalleryProvider = ({ children }: ImageGalleryProviderProps) => {
 				return setSingleImageUrl(target.dataset.id);
 			}
 			if (target?.classList.contains('gallery-item')) {
+				const id1 = target.dataset.src?.split('/file-upload/')[1].split('/')[0];
```

Comment:
What is this for?

Reply:
Currently, when sharing multiple files in one message, the preview always shows the very first image from the set, even if a different image is clicked. The preview should display the selected image correctly.

I've made adjustments to address this issue, ensuring that the correct image is displayed when clicked.

Reply:
Got it, we need to solve the root issue later, the issue happens here https://github.com/abhipatel0211/rocket.chat/blob/cdc0ff5ff9f82be59af8bdf4b4e70e478ebf129d/apps/meteor/client/components/message/variants/room/RoomMessageContent.tsx#L73 since we pass only the first file id to the attachment elements. We, probably, have to save the file id inside the attachment to use it there and use the first id passed as fallback (for the old records)